sql/src/main/java/io/crate/execution/dsl/phases/JoinPhase.java

```diff
@@ -25,65 +25,81 @@
 import com.google.common.base.MoreObjects;
 import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Iterables;
+import io.crate.execution.dsl.projection.Projection;
 import io.crate.expression.symbol.Symbol;
 import io.crate.expression.symbol.SymbolVisitors;
 import io.crate.expression.symbol.Symbols;
 import io.crate.planner.distribution.DistributionInfo;
 import io.crate.planner.node.dql.join.JoinType;
-import io.crate.execution.dsl.projection.Projection;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 
 import javax.annotation.Nullable;
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
 import java.util.UUID;
 
-public class NestedLoopPhase extends AbstractProjectionsPhase implements UpstreamPhase {
+public class JoinPhase extends AbstractProjectionsPhase implements UpstreamPhase {
 
     private final Collection<String> executionNodes;
     private final MergePhase leftMergePhase;
     private final MergePhase rightMergePhase;
     private final JoinType joinType;
+    private final ExecutionPhase.Type type;
 
     @Nullable
     private final Symbol joinCondition;
+    private final List<Symbol> leftJoinConditionInputs;
```

Comment:
I'm not sure if it's a good idea to add properties that are only relevant for certain types of this phase. Wouldn't it be cleaner to add a dedicated HashJoinPhase?

Reply:
Yeah, I agree now that I see it again.
But question, do you think would be nice to have a HashJoinPhase that inherits from NestedLoopPhase? or introduce abstract JoinPhase as supe rclass for both?

Reply:
IMO there should be 2 separate components (NLPhase and HJPhase) without any inheritance

Reply:
@andreidan I don't agree because then I need to have to Join execution plans...
https://github.com/crate/crate/pull/6903/files#diff-670704da4c49feaa815e6966562d085e